newsfragments/2734.changed.md

```diff
@@ -0,0 +1,2 @@
+`PyAny::cast_as()` and `Py::cast_as()` are now deprecated in favor of
```

Comment:
Maybe also have a `2734.added.md` just to make it clear that `Py::downcast` is new?